File: reference/CodeExamples/dotNetAppExamples/WCFDataServices/WCFDataServices/Person.cs
using System.Data.Services.Common;

namespace Db4oDoc.WCFDataService
{
    // #example: Add at least one key
    [DataServiceKey("Email")]
    public class Person
    {
        private string name;
        private string email;

        // Note that a default constructor is required for WCF
        public Person()
        {
        }

        public Person(string email, string name)
        {
            this.email = email;
            this.name = name;
        }

        public string Name
        {
            get { return name; }
            set { name = value; }
        }

        public string Email
        {
            get { return email; }
            set { email = value; }
        }

    }
    // #end example
}

File: reference/CodeExamples/java/src/com/db4odoc/pitfalls/activation/Person.java
package com.db4odoc.pitfalls.activation;

// #example: Person with a reference to the mother
class Person {
    private Person mother;
    private String name;
    public Person(String name) {
        this.mother = mother;
        this.name = name;
    }

    public Person(Person mother, String name) {
        this.mother = mother;
        this.name = name;
    }

    public Person mother() {
        return mother;
    }

    public String getName() {
        return name;
    }
}
// #end example
